### DISCUSSION TITLE
Pytanie dot. RELAY_IMPULSE  i REED_SWITCH

### DISCUSSION THREAD

**Discussion by paku2020** [lkankowski/arduino-multi-relay2#31] [Q&A]
Hej,
Chciałbym dopytać o funkcjonowanie RELAY_IMPULSE  i REED_SWITCH.

W konfiguracji mam takie coś:
`{5, 23, RELAY_TRIGGER_HIGH | RELAY_IMPULSE, -1, "Otwieranie bramy warsztat"},`
`{6, 22, RELAY_TRIGGER_HIGH | RELAY_IMPULSE, -1, "Otwieranie bramy wjazd"},`
Czyli w przypadku gdy w Domoticz kliknę na przełącznik "Otwieranie bramy warsztat"  złączy się przekaźnik nr 23 na 250ms i rozłączy?

Chciałem wykorzystać REED_SWITCH do kontaktronów.
Mam tak:
`{30, REED_SWITCH, 5, -1, -1, "Brama Warsztat-kontaktron"},`
`{32, REED_SWITCH, 6, -1, -1, "Brama Warsztat-kontaktron"}`
Czyli na pin 30,32 jest wpięty kontaktor pod arduino, w chwili złączeniach będzie status zamknięty. Prawda?
W konfiguracji trzeba podać pin przekaźnika. Czy można wpisać np. -1 ? Chyba że cokolwiek by nie było wpisane to część w której jest podany pin przekaźnika zostaje zignorowany ?

Ostatnie pytanie :)  
Czy w konfiguracji w ostatnich wierszach jest wymagany przecinek czy też nie ? Bo czasem zapominam dopisać a wydaje mi się że działa.

Pozdrawiam
Paweł

**Comment by lkankowski** [comment]
Zacznijmy od tego, jaki chcesz mieć efekt końcowy i wtedy dobierzemy rozwiązanie :)

Co do przecinków, to nie przeszkadzają jak występują za ostatnim elementem tablicy.

**Reply by paku2020** [reply]
Efekt dość prosty :)
Z domoticza chciałbym otwierać bramę  i chciałbym w nim widzieć stan z kontaktronu.
Wydaje mi się że ta konfiguracja jest chyba poprawna, ale wolałbym się upewnić

**Comment by lkankowski** [comment]
Zatem Twoja konfiguracja jest nieprawidłowa :)
Z niej wynika, że zmiana stanu kontaktronu, wywołuje zmianę przekaźnika, czyli zamykanie/otwieranie bramy.

Potrzebujesz wirtualnego przekaźnika, bo co do zasady, włączniki, a więc też kontaktrony nie są raportowane.
Proponuję tak:
```
...
{5, 23, RELAY_TRIGGER_HIGH | RELAY_IMPULSE, -1, "Otwieranie bramy warsztat"},
{6, 22, RELAY_TRIGGER_HIGH | RELAY_IMPULSE, -1, "Otwieranie bramy wjazd"},
{68, -1, 0, -1, "Brama Warsztat-kontaktron"},
{69, -2, 0, -1, "Brama Wjazd-kontaktron"},
...
{30, REED_SWITCH, 68, -1, -1, "Brama Warsztat-kontaktron"},
{32, REED_SWITCH, 69, -1, -1, "Brama Wjazd-kontaktron"}
...
```
Zamiast ID 68 i 69 ustal coś swojego.
Ujemna konfiguracja pinów oznacza "piny" wirtualne.

**Reply by paku2020** [reply]
A to chyba coś źle zrozumiałem w opisie :)
Dzięki :)

A co do wirtualnych -1,-2  itd
W całym konfigu mogą być wszędzie np -1 czy muszą być wirtualne unikalne  piny?

**Reply by lkankowski** [reply]
Wirtualne piny nie wpływają na siebie, więc mogą, ale lepiej dla porządku ustaw różne.

**Reply by paku2020** [reply]
Oki dziękuję :)
Przetestuję w takim razie to raz jeszcze

**Reply by paku2020** [reply]
Wszystko już wiem :) źle podchodziłem do tego i dlatego coś mi nie wychodziło.
Przy okazji zapytam jeszcze czy jest jakiś sposób lub opis gdzieś aby do sketchu dodać obsługę np dht22 ?

**Reply by lkankowski** [reply]
Temat dość często powraca, głównie na grupach FB, więc napiszę podobnie jak zwykle :)
Jeżeli nie ma bezpośredniego związku z funkcjonalnością główną (oświetlenie), a sensorami, to nie pchałbym tego na jedno urządzenie.
Oświetlenie jest krytyczne, a każda dodatkowa funkcjonalność, każdy dodatkowy kod zwiększa szansę na awarię - np. drobny błąd w bibliotece obsługi sensora w przypadku jego uszkodzenia mógłby w środku nocy unieruchomić oświetlenie w domu.
Tak więc doradzam jakieś inne małe i tanie Arduino do obsługi sensorów, aby zaoszczędzić sobie nerwów.

**Reply by paku2020** [reply]
W sumie może i racja
